common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -285,10 +285,11 @@ def get_html(self, view=STUDENT_VIEW):
         if getattr(self, 'video_speed_optimizations', True) and cdn_url:
             branding_info = BrandingInfoConfig.get_config().get(self.system.user_location)
 
-            for index, source_url in enumerate(sources):
-                new_url = rewrite_video_url(cdn_url, source_url)
-                if new_url:
-                    sources[index] = new_url
+            if self.edx_video_id and edxval_api:
```

Comment:
What is `edxval_api` here and why is it needed as part of the `if`?

Reply:
It is the video abstraction layer app, which we attempt to import at the beginning of this file.  If it is not present, then we cannot retrieve any metadata that is associated with video ids.